docs/walkthroughs/secure-ml-pipeline.md

```diff
@@ -0,0 +1,263 @@
+    hdfs@LOCAL
+    client@LOCAL
+    ```
+1.  Download the `keytab` file to a local directory.
+1.  Base64 encode the file with `base64 -w 0 {source} > {destination}` (we recommend using `base64` from GNU coreutils, if the version on your machine is different the `-w` flag may need to be replaced with the appropriate flag to disable wrapping.
+1.  Install the DC/OS Enterprise CLI with the command line: `dcos package install --yes dcos-enterprise-cli`   
+1.  Upload the base64 encoded file to the secret store: `dcos security secrets create __dcos_base64__keytab --value-file {destination}`.
+    *   Notice we have uploaded the secret to the root (will be `/__dcos_base64__keytab`)
+1.  You're now ready to install HDFS and Kafka. This can be done using the UI (DC/OS Catalog) or through the CLI as demonstrated here. First make an options.json containing at least (these are the same for both services):
+    ```json
+    {
+      "service": {
+        "kerberos": {
+          "enabled": "true",
+          "kdc": {
+            "hostname":  "<agent_hosting_kdc_or_dns>",
+            "port": "<port_to_communicate_with_kdc>"
+          },
+          "realm": "LOCAL",
+          "keytab_secret": "__dcos_base64__keytab"
+        },
+        "transport_encryption": {
+          "enabled": "true",
+          "allow_plaintext": "false"
+        }
+      }
+    }
+    ```
+    
+## Install Spark and prepare Java Authentication and Authorization Service (JAAS) file for Kafka
+1.  Install Spark including HDFS settings, using the instructions in [installing-secure-spark.md](). Using Kafka from Spark does not require any special installation.
+1.  Using Kerberos-secured Kafka from Spark requires a JAAS file to be available to the job, we use S3 to host the artifact. In this example use the following: 
+    ```
+    KafkaClient {
+        com.sun.security.auth.module.Krb5LoginModule required
+        useKeyTab=true
+        storeKey=true
+        keyTab="/mnt/mesos/sandbox/kafka-client.keytab"
+        useTicketCache=false
+        serviceName="kafka"
+        principal="client@LOCAL";
+    }; 
+    ```
+    This is also available in the [resources](https://github.com/mesosphere/spark-build/tree/master/tests/resources) of the test repo. Add these contents to a local file and copy it to S3 and make it publicly accessible. For the remainder of the tutorial we'll reference this location as `http://JAAS_ARTIFACT.conf` 
+
+## Run the example
+_All of the code for this example is in SpamHam.scala in the [resources](https://github.com/mesosphere/spark-build/tree/master/tests/resources) folder of this repo_
+1.  We need to upload the training data to S3, we use the file `SMSSpamCollection.txt` that can be found in the [resources](https://github.com/mesosphere/spark-build/tree/master/tests/resources) of the test repo. This artifact needs to be publicly available to your cluster, we will refer to this artifact as `http://SMS_DATA.txt`.
```

Comment:
why not hdfs?

Reply:
Unfortunately, at this time the Mesos fetcher will not download files from `hdfs` directly. To use HDFS you would need to put it behind an artifact server or use Web HDFS.